### PR TITLE
fix for #3223

### PR DESCRIPTION
**Pull request by byilmazturk**
added identifier checks for slaves while running checkconfig http://trac.buildbot.net/ticket/3223

### CODE REVIEW

File: master/buildbot/config.py [hunk 1]
```diff
@@ -541,6 +541,10 @@ def load_slaves(self, filename, config_dict):
             if sl.slavename in ("debug", "change", "status"):
                 msg = "slave name '%s' is reserved" % sl.slavename
                 error(msg)
+            
+            if not util.identifiers.isIdentifier(50, sl.slavename):
+                msg = "Buildslave name %r is not a 50-character identifier" % (sl.slavename,)
+                error(msg)
```
Comment (jaredgrubb) [inline master/buildbot/config.py]: This error makes it sound like it must be 50 chars, not something at most 50 chars... maybe "Buildslave name %r must be an identifier of at most 50 chars" or something?

Comment (byilmazturk) [inline master/buildbot/config.py]: Actually I copied error messages from https://github.com/buildbot/buildbot/blob/master/master/buildbot/data/buildslaves.py#L115 . I can change both of them, but I see that isIdentifier function does not control only length of slavename. Slavename can also break some other rules. May it be good to print all of those rules, if name breaks one of them?

Comment (sa2ajj) [inline master/buildbot/config.py]: Sounds like a good idea: guessing over a cryptic error message is not really a productive way of spending time. :smile:

Comment (djmitche) [inline master/buildbot/config.py]: You could also link to the docs..

**Comment by djmitche** [comment]
Any chance you'll have time to finish this up?

**Comment by byilmazturk** [comment]
sorry, I do not look and answer previous week because of holiday. I am going to check this now.

File: master/buildbot/config.py [hunk 2]
```diff
@@ -538,8 +539,19 @@ def load_slaves(self, filename, config_dict):
...
-            if sl.slavename in ("debug", "change", "status"):
-                msg = "slave name '%s' is reserved" % sl.slavename
+            def validate(slavename):
+                if slavename in ("debug", "change", "status"):
+                    yield "slave name '%s' is reserved" % slavename
+                if not isinstance(slavename, unicode):
+                    yield "slave name %r - is not a unicode string" % slavename
+                if not util_identifiers.ident_re.match(slavename):
+                    yield "slave name %r - is not an identifier" % slavename
+                if len(slavename) < 1:
```
Comment (sa2ajj) [inline master/buildbot/config.py]: should not this be 'if not slavename:` instead?

**Comment by sa2ajj** [comment]
LGTM otherwise

**Comment by tardyp** [comment]
:+1:

**Pull request merged**